### PR TITLE
chore: updated login default theme

### PR DESCRIPTION
**Pull request by ashraf706**
Log in flow default theme has been updated as per new design guideline.
The github issue 8042 contains further design details.
Single main.css has been used across all html files.

###  All the html files have been updated as per the design doc. The design doc is available in this link:
https://www.figma.com/file/PKtV6NcsNzGBuzV3cmGHJj/Custom-login?node-id=590%3A4557

### CODE REVIEW

**Review by leleueri — CHANGES_REQUESTED** [review]
Here is some feedbacks:

* WebAuthn / Passwordless : the "Back to sign in" arrow is missing.
image

* MFA enrollment, when the method is selected, the link "Back to singin" should be "back to methods" 
image

* Consent screen, a space is missing "accordance to" 
image

* The ResetPassword form show only one password field, but the figma mockup show 2 fields, password and confirm password

* The UserRegistration form can't be submitted if there are no password policy

* I think the login.css file can be remove

File: gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-core/src/main/resources/webroot/assets/css/login_flow.css [hunk 1]
```diff
@@ -0,0 +1,630 @@
+.container {
+    --topBorderColor: #6A4FF7;
```
Comment (tcompiegne) [inline gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-core/src/main/resources/webroot/assets/css/login_flow.css:2]: A good practice is not define all the variables in a dedicated section and reuse them after : 

```
:root {
  --topBorderColor: #6A4FF7;
  --width: 222px;
  --backgroundColor: #6A4FF7;
  --textColor: #FFFFFF;
}
```

and use them in the css file.

That way the only part of the custom theme will be to generate a custom `:root { }` css element 

```

File: gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-core/src/main/resources/webroot/views/forgot_password.html [hunk 2]
```diff
@@ -8,10 +8,12 @@
...
     <!-- CSS -->
-    <link rel="stylesheet" th:href="@{assets/material/material.blue_grey-blue.min.css}">
-    <link rel="stylesheet" th:href="@{assets/material/material.icons.css}">
-    <link rel="stylesheet" th:href="@{assets/font-awesome/css/font-awesome.min.css}">
-    <link rel="stylesheet" th:href="@{assets/css/forgot_password.css}">
+    <link rel="stylesheet" th:href="@{assets/css/login_flow.css}">
+
+    <link rel="preconnect" href="https://fonts.googleapis.com">
+    <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
+    <link href="https://fonts.googleapis.com/css2?family=Roboto:wght@500;700;900&display=swap" rel="stylesheet">
```
Comment (tcompiegne) [inline gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-core/src/main/resources/webroot/views/forgot_password.html:15]: I'm not a huge fan of calling external assets (for enterprise proxy issue mainly).

Is it possible to use the font-face CSS feature ? something like : 

```
in the main.css file : 

 /* get the required local files */
    @font-face {
      font-family: 'Roboto';
      src: url('roboto.eot'); /* IE9 Compat Modes */
      src: url('roboto.eot?#iefix') format('embedded-opentype'), /* IE6-IE8 */
      url('roboto.woff') format('woff'), /* Modern Browsers */
      url('roboto.ttf')  format('truetype'), /* Safari, Android, iOS */
      url('roboto.svg#svgFontName') format('svg'); /* Legacy iOS */
    }
    
    /* use the font */
    body {
      font-family: 'Roboto', sans-serif;
      font-size: 48px;
    }
```
More information here : https://stackoverflow.com/questions/18178867/how-can-i-use-googles-roboto-font-on-a-website

```

File: gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-core/src/main/resources/webroot/assets/css/login_flow.css [hunk 3]
```diff
@@ -0,0 +1,630 @@
...
+    line-height: 16px;
+    text-align: center;
+    color: #25213A;
+    flex: none;
+    order: 1;
+    flex-grow: 0;
+}
+
+.header-description {
+    font-family: 'Roboto', sans-serif;
```
Comment (tcompiegne) [inline gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-core/src/main/resources/webroot/assets/css/login_flow.css:76]: font-family should only be declared once at body level : 

```
body {
  font-family: 'Roboto', sans-serif;
}
```

**Review by tcompiegne — CHANGES_REQUESTED** [review]

File: gravitee-am-management-api/gravitee-am-management-api-rest/src/main/resources/views/login.html [hunk 4]
```diff
@@ -12,7 +12,6 @@
     <link rel="stylesheet" th:href="@{assets/material/material.icons.css}">
     <link rel="stylesheet" th:href="@{assets/font-awesome/css/font-awesome.min.css}">
     <link rel="stylesheet" th:href="@{assets/LibreFranklin/font-libre-franklin.css}">
-    <link rel="stylesheet" th:href="@{assets/css/login.css}">
```
[the remark this answers is not in the document]
Comment (ashraf706) [reply gravitee-am-management-api/gravitee-am-management-api-rest/src/main/resources/views/login.html:15]: reverted, login.css added again.

File: gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-core/src/main/resources/webroot/assets/css/login_flow.css [hunk 5]
```diff
@@ -0,0 +1,630 @@
+.container {
```
[the remark this answers is not in the document]
Comment (ashraf706) [reply gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-core/src/main/resources/webroot/assets/css/login_flow.css]: done

File: gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-core/src/main/resources/webroot/assets/css/login_flow.css (hunk 1 above)
Comment (ashraf706) [reply gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-core/src/main/resources/webroot/assets/css/login_flow.css]: done, :root {} added

File: gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-core/src/main/resources/webroot/views/forgot_password.html (hunk 2 above)
Comment (ashraf706) [reply gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-core/src/main/resources/webroot/views/forgot_password.html]: done

File: gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-core/src/main/resources/webroot/assets/css/login_flow.css (hunk 3 above)
Comment (ashraf706) [reply gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-core/src/main/resources/webroot/assets/css/login_flow.css]: done

**Review by tcompiegne — CHANGES_REQUESTED** [review]
Do we still need the login.css and password_validation.css files in the gateway side ?

**Review by tcompiegne — CHANGES_REQUESTED** [review]
- Missing placeholder "Email" in the forgot password page
- Missing underline for the "Sign in" link in the register page
- "Back to sign in" link in the forgot password confirmation page is blue instead of underline and black
- Missing underline for the "Back to sign in" link in the register confirmation page

**Review by tcompiegne — CHANGES_REQUESTED** [review]
- Missing dot in front of all the recovery code list
- Recovery code list should be centered in the page

The second part difficult and I couldn't do it nicely. I am looking into it. **Now much better**

**Comment by ashraf706** [comment]
> * Missing placeholder "Email" in the forgot password page
> * Missing underline for the "Sign in" link in the register page
> * "Back to sign in" link in the forgot password confirmation page is blue instead of underline and black
> * Missing underline for the "Back to sign in" link in the register confirmation page

Done

File: gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-core/src/main/resources/webroot/assets/css/login.css [hunk 6]
```diff
@@ -97,11 +97,11 @@ a, a:hover, a:focus {
...
 
-.social-login-buttons {
+.social-button {
     margin-top: 10px;
 }
 
-.social-login-buttons a {
+.social-button a {
     margin-bottom: 10px;
 }
```
Comment (tcompiegne) [inline gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-core/src/main/resources/webroot/assets/css/login.css:106]: Undo changes

File: gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-core/src/main/resources/webroot/assets/css/password_validation.css [hunk 7]
```diff
@@ -1,10 +1,10 @@
-#passwordSettings {
+/*#passwordSettings {
     background: #f1f1f1;
     color: #000;
     position: relative;
     padding: 20px;
     margin: 10px 0 10px 0;
-}
+}*/
 #passwordSettings p {
```
Comment (tcompiegne) [inline gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-core/src/main/resources/webroot/assets/css/password_validation.css:8]: Undo changes

**Review by tcompiegne — CHANGES_REQUESTED** [review]

File: gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-core/src/main/resources/webroot/assets/css/login.css (hunk 6 above)
Comment (ashraf706) [reply gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-core/src/main/resources/webroot/assets/css/login.css]: done

File: gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-core/src/main/resources/webroot/assets/css/password_validation.css (hunk 7 above)
Comment (ashraf706) [reply gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-core/src/main/resources/webroot/assets/css/password_validation.css]: done

**Review by tcompiegne — APPROVED** [review]

**Pull request merged**